File: src/main/java/it/unitn/dbtrento/spark/utils/FieldDelimiter.java
package it.unitn.dbtrento.spark.utils;

public enum FieldDelimiter {
  COMMA(','), TAB('\t'), SPACE(' '), COLON(':'), SEMICOLON(';');

  FieldDelimiter(char fieldDelimiter){
    this.fieldDelimiter = fieldDelimiter;
  }

  private final char fieldDelimiter;

  public char getFieldDelimiter() {
    return fieldDelimiter;
  }
}

File: src/main/scala/eu/unitn/disi/db/spark/utils/FieldDelimiter.scala
package eu.unitn.disi.db.spark.utils

object FieldDelimiter extends Enumeration {

  protected case class Val(fieldDelimiter: Char) extends super.Val {
    def getFieldDelimiter: Char = fieldDelimiter
  }

  implicit def valueToFieldDelimiterVal(fieldDelimiter: Value) =
    fieldDelimiter.asInstanceOf[Val]

  val COMMA = Val(',')
  val TAB = Val('\t')
  val SPACE = Val(' ')
  val COLON = Val(':')
  val SEMICOLON = Val(';')
}
